src/sos/preview.py

```diff
@@ -230,11 +230,45 @@ def preview_md(filename, kernel=None, style=None):
     return {'text/html': html, 'text/plain': text}
 
 def preview_dot(filename, kernel=None, style=None):
+    from os import listdir, remove
+    from os.path import isfile, join
     from graphviz import Source
+    import tempfile
+    import shutil
     with open(filename) as dot:
-        src = Source(dot.read())
-    src.format='png'
-    outfile = src.render()
-    with open(outfile, 'rb') as content:
-        data = content.read()
-    return {'image/png': base64.b64encode(data).decode('ascii') }
+        fileNameElement = "sosDotFilesPng"
+        with tempfile.TemporaryDirectory() as tempDirectory:
```

Comment:
No need for nested `with` since you already have `data` read from `outfile`, right?

Reply:
You mean change it into `tempDirectory = tempfile.TemporaryDirectory().name`?
In this case, do I need to remove the TemporaryDirectory?